I have exactly the same setup in my reverse proxy (I'm using nginx k8s ingress, but the idea is the same).

The problem is, that mobile application needs to get this token in a first place. For clarity - I'm using [paperless-mobile](https://github.com/astubenbord/paperless-mobile). In this app, authentication is done this way:

- Firstly, app pings https://paperless.mydomain.com/api/ to check if domain is alive. It does it without any token - just naked `GET` request. When `ENABLE_HTTP_REMOTE_USER` is on, and `/api/*` is "chain-no-auth", it raises `500` error:

```
$ curl https://paperless.domain.com/api/
<!doctype html>
<html lang="en">
<head>
  <title>Server Error (500)</title>
</head>
<body>
  <h1>Server Error (500)</h1><p></p>
</body>
</html>
```

and in logs I see:

```
[2023-06-03 10:15:02,989] [ERROR] [django.request] Internal Server Error: /api/
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/asgiref/sync.py", line 486, in thread_handler
    raise exc_info[1]
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/exception.py", line 43, in inner
    response = await get_response(request)
  File "/usr/local/lib/python3.9/site-packages/django/core/handlers/base.py", line 253, in _get_response_async
    response = await wrapped_callback(
  File "/usr/local/lib/python3.9/site-packages/asgiref/sync.py", line 448, in __call__
    ret = await asyncio.wait_for(future, timeout=None)
  File "/usr/local/lib/python3.9/asyncio/tasks.py", line 442, in wait_for
    return await fut
  File "/usr/local/lib/python3.9/site-packages/asgiref/current_thread_executor.py", line 22, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.9/site-packages/asgiref/sync.py", line 490, in thread_handler
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/views/decorators/csrf.py", line 55, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/views/generic/base.py", line 103, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 497, in dispatch
    self.initial(request, *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 414, in initial
    self.perform_authentication(request)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/views.py", line 324, in perform_authentication
    request.user
  File "/usr/local/lib/python3.9/site-packages/rest_framework/request.py", line 227, in user
    self._authenticate()
  File "/usr/local/lib/python3.9/site-packages/rest_framework/request.py", line 380, in _authenticate
    user_auth_tuple = authenticator.authenticate(self)
  File "/usr/local/lib/python3.9/site-packages/rest_framework/authentication.py", line 230, in authenticate
    user = authenticate(request=request, remote_user=request.META.get(self.header))
  File "/usr/local/lib/python3.9/site-packages/django/views/decorators/debug.py", line 42, in sensitive_variables_wrapper
    return func(*func_args, **func_kwargs)
  File "/usr/local/lib/python3.9/site-packages/django/contrib/auth/__init__.py", line 89, in authenticate
    user_login_failed.send(
  File "/usr/local/lib/python3.9/site-packages/django/dispatch/dispatcher.py", line 176, in send
    return [
  File "/usr/local/lib/python3.9/site-packages/django/dispatch/dispatcher.py", line 177, in <listcomp>
    (receiver, receiver(signal=self, sender=sender, **named))
  File "/usr/src/paperless/src/paperless/signals.py", line 31, in handle_failed_login
    f"Login failed for user `{credentials['username']}`"
KeyError: 'username'
```

- Second step, for mobile app is to send username and password to `/api/token/` endpoint to get the token. It returns the same 500 and the same exception.

```bash
curl -X POST -d 'username=user&password=pass' https://paperless.domain.com/api/token/
```

That's why disabling auth middlewares on these two helps. 

For reference, here are my env variables:

```
PAPERLESS_ALLOWED_HOSTS: "*"
PAPERLESS_CONSUMER_POLLING: 30 # in seconds, ionotify doesn't work on ceph
PAPERLESS_DBHOST: postgresql-ha.storage.svc.cluster.local
PAPERLESS_DBNAME: paperless
PAPERLESS_DBPORT: 5432
PAPERLESS_DBSSLMODE: require
PAPERLESS_ENABLE_HTTP_REMOTE_USER: "true" # <----
PAPERLESS_HTTP_REMOTE_USER_HEADER_NAME: "HTTP_REMOTE_USER" # <----
PAPERLESS_OCR_LANGUAGE: "eng+pol"
PAPERLESS_PORT: 8000
PAPERLESS_TIKA_ENABLED: 1
PAPERLESS_TIKA_ENDPOINT: "http://tika.apps.svc.cluster.local:9998"
PAPERLESS_TIKA_GOTENBERG_ENDPOINT: "http://gotenberg.apps.svc.cluster.local:3000"
PAPERLESS_TIME_ZONE: "Europe/Warsaw"
PAPERLESS_URL: "https://paperless.domain.com"
```

I'll try to do some experimentations, and patch this `500` issue, to see if it helps, maybe that's the culrpit 🤔

--- a/src/paperless/auth.py
+++ b/src/paperless/auth.py
@@ -1,6 +1,6 @@
 from django.conf import settings
 from django.contrib import auth
-from django.contrib.auth.middleware import RemoteUserMiddleware
+from django.contrib.auth.middleware import PersistentRemoteUserMiddleware
 from django.contrib.auth.models import User
 from django.utils.deprecation import MiddlewareMixin
 from rest_framework import authentication
@@ -37,7 +37,7 @@ class AngularApiAuthenticationOverride(authentication.BaseAuthentication):
             return None
 
 
-class HttpRemoteUserMiddleware(RemoteUserMiddleware):
+class HttpRemoteUserMiddleware(PersistentRemoteUserMiddleware):
     """This class allows authentication via HTTP_REMOTE_USER which is set for
     example by certain SSO applications.
     """

--- a/src/paperless/signals.py
+++ b/src/paperless/signals.py
@@ -12,21 +12,21 @@ def handle_failed_login(sender, credentials, request, **kwargs):
     client_ip, _ = ipware.get_client_ip(
         meta=request.META,
     )
-    username = credentials.get("username") or "anonymous"
+    username = credentials.get("username")
+    log_output = (
+        "No authentication provided"
+        if username is None
+        else f"Login failed for user `{username}`"
+    )
 
     if client_ip is None:
-        logger.info(
-            f"Login failed for user `{username}`. Unable to determine IP address.",
-        )
+        log_output += ". Unable to determine IP address."
     else:
         if client_ip.is_global:
             # We got the client's IP address
-            logger.info(
-                f"Login failed for user `{username}` from IP `{client_ip}.`",
-            )
+            log_output += f" from IP `{client_ip}.`"
         else:
             # The client's IP address is private
-            logger.info(
-                f"Login failed for user `{username}`"
-                f" from private IP `{client_ip}.`",
-            )
+            log_output += f" from private IP `{client_ip}.`"
+
+    logger.info(log_output)

--- a/src/paperless/signals.py
+++ b/src/paperless/signals.py
@@ -12,22 +12,21 @@ def handle_failed_login(sender, credentials, request, **kwargs):
     client_ip, _ = ipware.get_client_ip(
         meta=request.META,
     )
+    username = credentials.get("username") or "anonymous"
 
     if client_ip is None:
         logger.info(
-            f"Login failed for user `{credentials['username']}`."
-            " Unable to determine IP address.",
+            f"Login failed for user `{username}`. Unable to determine IP address.",
         )
     else:
         if client_ip.is_global:
             # We got the client's IP address
             logger.info(
-                f"Login failed for user `{credentials['username']}`"
-                f" from IP `{client_ip}.`",
+                f"Login failed for user `{username}` from IP `{client_ip}.`",
             )
         else:
             # The client's IP address is private
             logger.info(
-                f"Login failed for user `{credentials['username']}`"
+                f"Login failed for user `{username}`"
                 f" from private IP `{client_ip}.`",
             )

--- a/src/paperless/tests/test_signals.py
+++ b/src/paperless/tests/test_signals.py
@@ -12,6 +12,26 @@ class TestFailedLoginLogging(TestCase):
             "username": "john lennon",
         }
 
+    def test_unauthenticated(self):
+        """
+        GIVEN:
+            - Request with no authentication provided
+        WHEN:
+            - Request provided to signal handler
+        THEN:
+            - Unable to determine logged for unauthenticated user
+        """
+        request = HttpRequest()
+        request.META = {}
+        with self.assertLogs("paperless.auth") as logs:
+            handle_failed_login(None, {}, request)
+            self.assertEqual(
+                logs.output,
+                [
+                    "INFO:paperless.auth:No authentication provided. Unable to determine IP address.",
+                ],
+            )
+
     def test_none(self):
         """
         GIVEN:

